demo/full/scripts/controllers/Player.tsx

```diff
@@ -254,13 +262,66 @@ function Player(): JSX.Element {
     [playerModule],
   );
 
+  const handleContentConfigChange = useCallback((contentConfigArg: ContentConfig) => {
+    setContentConfig(contentConfigArg);
+  }, []);
+
+  const generatedURL = React.useMemo<null | string>(() => {
+    if (contentConfig === null) {
+      return null;
+    }
+    return generateURLForConfig({
+      contentConfig,
+      loadVideoConfig: loadVideoOpts,
+      playerConfig: playerOpts,
+      demoConfig: { reactiveURL: isReactiveURLEnabled },
+    });
+  }, [contentConfig, loadVideoOpts, playerOpts, isReactiveURLEnabled]);
+
+  const [sharableURL, setSharableURL] = React.useState("");
+  React.useEffect(() => {
+    // still update URL if the last change was editing "isReactiveURLEnabled"
+    // value, so the "isReactiveURLEnabled" can be stored in the URL
+    const isEnabled = previousIsReactiveURLEnabled || isReactiveURLEnabled;
+    if (generatedURL && isEnabled) {
+      setSharableURL(generatedURL);
+      window.location.href = generatedURL;
+    }
+  }, [generatedURL, isReactiveURLEnabled]);
+
+  React.useEffect(() => {
+    const parsedHash = parseHashInURL(decodeURI(location.hash));
+    if (parsedHash !== null) {
+      const { loadVideoConfig, playerConfig, disableReactiveURL } = parsedHash;
+      if (typeof loadVideoConfig === "string") {
+        // maybe we want to check the shape of the object to
+        // be error prone instead of casting?
```

Comment:
:man_shrugging:

Reply:
I have made some changes here, JSON.parse() is not safe if the string is not convertible to a JS object. Therefore I wrapped it in a try/catch.
I have added some checks on the shape of the parsed object, there are quite simple as they only check the "typeof", that would be enough. I didn't went as deep as checking if values are respecting the typescript intersection (maybe it's preferable to use a validating library like Yup or Joi if we want to make such checks?)

I have also handled the case where we stringify/Parse "Infinity'